cirq-core/cirq/sim/density_matrix_simulator.py

```diff
@@ -367,6 +367,11 @@ def final_density_matrix(self) -> np.ndarray:
             size = np.prod(protocols.qid_shape(self), dtype=np.int64)
             tensor = self._get_merged_sim_state().target_tensor
             self._final_density_matrix = np.reshape(tensor.copy(), (size, size))
+            if abs(np.trace(self._final_density_matrix) - 1) < 1e-6:
```

Comment:
Can you store the trace in a local var so it doesn't need to be recomputed below?